pkg/aggregator/demultiplexer_agent_test.go

```diff
@@ -153,15 +156,19 @@ type DemultiplexerAgentTestDeps struct {
 	OrchestratorFwd orchestratorforwarder.Component
 	EventPlatform   eventplatform.Component
 	Compressor      compression.Component
+	Tagger          tagger.Component
 }
 
 func createDemultiplexerAgentTestDeps(t *testing.T) DemultiplexerAgentTestDeps {
+	taggerComponent := fxutil.Test[tagger.Mock](t, taggerimpl.MockModule())
+
 	return fxutil.Test[DemultiplexerAgentTestDeps](
 		t,
 		defaultforwarder.MockModule(),
 		core.MockBundle(),
 		orchestratorimpl.MockModule(),
 		eventplatformimpl.MockModule(),
 		compressionimpl.MockModule(),
+		fx.Provide(func() tagger.Component { return taggerComponent }),
```

Comment:
Unfortunately the `MockModule` internally uses other mocks, such as config mock module that is also used by the `core.MockModule` so fx complains because we use the same module twice. That is why I had to create the tagger outside